packages/browser/src/integrations/globalhandlers.ts

```diff
@@ -78,21 +78,51 @@ export class GlobalHandlers implements Integration {
     }
   }
 
-  /** JSDoc */
+  /**
+   * This function creates an SentryEvent from an TraceKitStackTrace.
+   *
+   * @param stacktrace TraceKitStackTrace to be converted to an SentryEvent.
+   */
   private eventFromGlobalHandler(stacktrace: TraceKitStackTrace): SentryEvent {
     const event = eventFromStacktrace(stacktrace);
-    return {
+
+    const data: { [key: string]: string } = {
+      mode: stacktrace.mode,
+    };
+
+    if (stacktrace.message) {
+      data.message = stacktrace.message;
+    }
+
+    if (stacktrace.name) {
+      data.name = stacktrace.name;
+    }
+
+    const newEvent: SentryEvent = {
       ...event,
       exception: {
         ...event.exception,
         mechanism: {
-          data: {
-            mode: stacktrace.mode,
-          },
+          data,
           handled: false,
           type: stacktrace.mechanism,
         },
       },
     };
+
+    let type = 'Error';
```

Comment:
I'd change the name to be more precise, something like `fallbackType`.

```js
const fallbackType = stacktrace.mechanism === 'onunhandledrejection' ? 'UnhandledPromiseRejection' : 'Error';
const fallbackValue = typeof stacktrace.original !== 'undefined' ? `${stacktrace.original}` : '';  
```

Simple `if (stacktrace.original)` check would fail for `reject(0)`.